src/Geocoder/Provider/Chain.php

```diff
@@ -38,6 +40,9 @@ public function geocode($address)
     {
         $exceptions = [];
         foreach ($this->providers as $provider) {
+            if ($provider instanceof LocaleAwareProvider && $this->getLocale() !== null){
+                $provider->setLocale($this->getLocale());
```

Comment:
This can lead to an logical error, see this example:

```php
$bing = new BingMaps($http, 'secret', 'de_DE'); // note the locale here
$chain = new Chain([$bing]);

$geocoder = new Geocoder();
$geocoder->registerProvider($bing);
$geocoder->registerProvider($chain);

$geocoder->geocode('Paris'); // Will now use de_DE locale
$chain->setLocale('en_US') // Chain now use en_US
$geocoder->use('chain')->geocoder('Paris'); // I get en_US results

$geocoder->geocode('Paris'); // I expect the locale is de_DE, but it will be en_US because the chain provider will override them.
```

I would suggest to clone the provider or set the original locale back.